apps/condo/migrations/20220531154236-0132_organization_tin_alter_organizationhistoryrecord_tin.js

```diff
@@ -0,0 +1,39 @@
+// auto generated by kmigrator
+// KMIGRATOR:0132_organization_tin_alter_organizationhistoryrecord_tin:IyBHZW5lcmF0ZWQgYnkgRGphbmdvIDQuMC40IG9uIDIwMjItMDUtMzEgMTA6NDIKCmZyb20gZGphbmdvLmRiIGltcG9ydCBtaWdyYXRpb25zLCBtb2RlbHMKCgpjbGFzcyBNaWdyYXRpb24obWlncmF0aW9ucy5NaWdyYXRpb24pOgoKICAgIGRlcGVuZGVuY2llcyA9IFsKICAgICAgICAoJ19kamFuZ29fc2NoZW1hJywgJzAxMzFfcmVtb3ZlX3RpY2tldGNoYW5nZV93YXRjaGVyc2Rpc3BsYXluYW1lc2Zyb21fYW5kX21vcmUnKSwKICAgIF0KCiAgICBvcGVyYXRpb25zID0gWwogICAgICAgIG1pZ3JhdGlvbnMuQWRkRmllbGQoCiAgICAgICAgICAgIG1vZGVsX25hbWU9J29yZ2FuaXphdGlvbicsCiAgICAgICAgICAgIG5hbWU9J3RpbicsCiAgICAgICAgICAgIGZpZWxkPW1vZGVscy5UZXh0RmllbGQoYmxhbms9VHJ1ZSwgbnVsbD1UcnVlKSwKICAgICAgICApLAogICAgICAgIG1pZ3JhdGlvbnMuQWx0ZXJGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0nb3JnYW5pemF0aW9uaGlzdG9yeXJlY29yZCcsCiAgICAgICAgICAgIG5hbWU9J3RpbicsCiAgICAgICAgICAgIGZpZWxkPW1vZGVscy5UZXh0RmllbGQoYmxhbms9VHJ1ZSwgbnVsbD1UcnVlKSwKICAgICAgICApLAogICAgXQo=
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Add field tin to organization
+--
+ALTER TABLE "Organization" ADD COLUMN "tin" text NULL;
+--
+-- Alter field tin on organizationhistoryrecord
+--
+ALTER TABLE "OrganizationHistoryRecord" ALTER COLUMN "tin" TYPE text USING "tin"::text;
+--
+-- Fill tin field with exists metadata
+--
+UPDATE "Organization" SET "tin" = "meta"::json->>'inn';
+UPDATE "OrganizationHistoryRecord" SET "tin" = "meta"::json->>'inn';
```

Comment:
it's a bad ided becouse `"meta"::json` have lotes of invalid data. Your field should be clean and trule validated

Reply:
Then we will need to write single script in binary folder to:
1. Fetch all organizations
2. For each organization validate tin and set it to new field
3. Get some statistics how much invalid and duplicated tins we have

Reply:
Bin script added